src/server/snap.rs

```diff
@@ -86,17 +85,16 @@ fn send_snap(mgr: SnapManager, addr: SocketAddr, data: ConnData) -> Result<()> {
     let snap = data.msg.get_raft().get_message().get_snapshot();
     let key = try!(SnapKey::from_snap(&snap));
     mgr.wl().register(key.clone(), SnapEntry::Sending);
-    let snap_file = box_try!(mgr.rl().get_snap_file(&key, true));
-    defer!({
-        snap_file.delete();
+    let mut f = box_try!(mgr.rl().get_send_snapshot_file(&key));
```

Comment:
I prefer keeping the `defer`, otherwise it maybe very easy to forget delete the file when doing refactor in the future.